homeassistant/components/iaqualink/climate.py

```diff
@@ -103,7 +103,14 @@ def max_temp(self) -> int:
     @property
     def target_temperature(self) -> float:
         """Return the current target temperature."""
-        return float(self.dev.state)
+        if not isinstance(self.dev.state, str) or self.dev.state == "":
+            return 0.0
+
+        try:
+            return float(self.dev.state)
+        except ValueError:
+            _LOGGER.debug("Unknown temperature: '%s'", self.dev)
```

Comment:
We don't allow side effects like logging in entity state properties.

Reply:
Updated to remove logging from state changes.